File: Lib/Telerik/Examples/QuickStart/Sparkline/FirstLook/ExampleViewModel.vb

Imports System
Imports System.Collections.Generic
Imports System.ComponentModel
Imports System.Linq
Imports System.Text
Imports System.Threading

Namespace Telerik.Examples.WinControls.Sparkline.FirstLook
    Public Class ExampleViewModel
        Private _callees As List(Of CalleeViewModel)
        Private _calls As BindingList(Of CallData)
        Private rand As Random

        Public Sub New()
            Me._callees = Me.GetCalleeViewModels()
            Me._calls = New BindingList(Of CallData)()
            Me.rand = New Random()
            Me.GenerateCalls()
        End Sub

        Public ReadOnly Property Callees As List(Of CalleeViewModel)
            Get
                Return Me._callees
            End Get
        End Property

        Public ReadOnly Property Calls As BindingList(Of CallData)
            Get
                Return Me._calls
            End Get
        End Property

        Public Sub GenerateCalls()
            Dim callData As CallData
            Dim dateCall As DateTime = New DateTime(2010, 12, 12, 10, 0, 0)

            For i As Integer = 0 To 24 - 1
                callData = New CallData With {
                    .Duration = New TimeSpan(0, Me.rand.[Next](0, 100), Me.rand.[Next](1, 59)),
                    .HoldTime = New TimeSpan(0, Me.rand.[Next](0, 100), Me.rand.[Next](1, 59)),
                    .Abandonments = Me.rand.[Next](0, 50),
                    .TimeInverval = dateCall
                }

                dateCall = dateCall.AddHours(1)
                Me.calls.Add(callData)
            Next
        End Sub

        Private Function GetCalleeViewModels() As List(Of CalleeViewModel)
            Dim callees As List(Of CalleeViewModel) = New List(Of CalleeViewModel)()
            callees.Add(New CalleeViewModel("Sam"))
            callees.Add(New CalleeViewModel("John"))
            callees.Add(New CalleeViewModel("Mike"))
            Return callees
        End Function
    End Class
End Namespace

File: Lib/Telerik/Examples/QuickStart/Sparkline/FirstLook/ExampleViewModel.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Linq;
using System.Text;
using System.Threading;

namespace Telerik.Examples.WinControls.Sparkline.FirstLook
{
    public class ExampleViewModel
    {
        private List<CalleeViewModel> callees;
        private BindingList<CallData> calls;
        private Random rand;

        public ExampleViewModel()
        {
            this.callees = this.GetCalleeViewModels();
            this.calls = new BindingList<CallData>();
            this.rand = new Random();
            this.GenerateCalls();
        }

        public List<CalleeViewModel> Callees
        {
            get
            {
                return this.callees;
            }
        }

        public BindingList<CallData> Calls
        {
            get
            {
                return this.calls;
            }
        }

        public void GenerateCalls()
        {
            CallData callData;
            DateTime date = new DateTime(2010, 12, 12, 10, 0, 0);
            for (int i = 0; i < 24; i++)
            {
                callData = new CallData
                {
                    Duration = new TimeSpan(0, this.rand.Next(0, 100), this.rand.Next(1, 59)),
                    HoldTime = new TimeSpan(0, this.rand.Next(0, 100), this.rand.Next(1, 59)),
                    Abandonments = this.rand.Next(0, 50),
                    TimeInverval = date
                };

                date = date.AddHours(1);

                this.calls.Add(callData);
            }
        }

        private List<CalleeViewModel> GetCalleeViewModels()
        {
            List<CalleeViewModel> callees = new List<CalleeViewModel>();
            callees.Add(new CalleeViewModel("Sam"));
            callees.Add(new CalleeViewModel("John"));
            callees.Add(new CalleeViewModel("Mike"));

            return callees;
        }
    }
}
